src/db/emoji_logging.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::*;
+
+use super::Db;
+
+use serenity::model::id::EmojiId;
+
+pub struct EmojiData {
+    //You could store the emoji itself in here
+    //    emoji: Emoji,
+    reactions: u64,
+    in_text: u64,
+}
+
+impl Default for EmojiData {
+    fn default() -> Self {
+        EmojiData {
+            reactions: 0,
+            in_text: 0,
+        }
+    }
+}
+
+impl Db {
+    pub async fn increment_emoji_reaction(
+        &self,
+        count: u64,
+        emoji_name: impl AsRef<str>,
+        emoji: &EmojiId,
+    ) -> Result<EmojiData> {
+        let mut data = self.get_emoji_usage(emoji).await?;
+        let mut conn = self.pool.acquire().await?;
+        let emoji_str = emoji_name.as_ref();
+        data.reactions += count;
+        let num = data.reactions as i64;
+        let id = emoji.0 as i64;
+        sqlx::query!("insert into emojis (emoji_id, emoji_name, reaction_usage) values (?1, ?2, ?3) on conflict(emoji_id) do update set reaction_usage=?3", id, emoji_str,num).execute(&mut conn).await?;
+        Ok(data)
```

Comment:
For these, SQL does have update increment stuff - u can `update emojis set reaction_usage=reaction_usage+1` (or sth like that). that'D be a lot cleaner than first querying the db, then adding the number manually, then setting (also leave less room for race-conditions, as this currently would definitely have data-races, - not that that matters all that much here)